Automattic/sensei
# Add new sensei patterns and separate block patterns logic

Fixes #5214
Fixes #5136

### Changes proposed in this Pull Request

* It adds the new course and lesson patterns.
* Patterns generic logic was already extracted to a new class in order to avoid a big refactor when we introduce new patterns not related to the editor wizard.
* Patterns were moved to new files, similar to WP core and TwentyTwenty Two. The main reason is that they are big.
* The filter to list patterns in the editor wizard was updated to block type `"sensei-lms/post-content"` (similar approach to the [core implementation](https://github.com/WordPress/gutenberg/pull/40034/files#diff-bec7de12ed6f642d1ca37c36d37fc67d301f34c30f9004072c219189fed18369R26)). The main reason is that probably we'll add other patterns in the future to the Sensei LMS category.

### Known issues

* The colors don't work well in some themes: https://github.com/Automattic/sensei/issues/5218

### Testing instructions


* Create a new course, and make sure you see the new patterns to use.
* Select them, and make sure they work properly.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/171296742-7f44ddc1-ba1c-4bff-940b-baa57e234f18.mov
